server/model/Publication.js

```diff
@@ -94,5 +92,25 @@ publicationSchema.post('save', (err, doc, next) => {
   return next(err);
 });
 
+var Publication = mongoose.model('Publication', publicationSchema);
+
+Publication.collection.dropIndexes();
+
+Publication.collection.createIndex(
+  {
+   videoID: "text",
+   ownerUsername: "text",
+   artist: "text",
+   title: "text"
+  },
+  {
+    weights: {
```

Comment:
Não entendi o por que disso.

Reply:
Pra possibilitar busca por vários campos, igual ao youtube.

Reply:
Eu defino os indexes que a busca deve usar e o peso pra cada um dos campos

Reply:
O drop e o create é por causa de um "bug" no mongo, que se você já tem index criado ele não atualiza mais. Tem que remover e criar novamente.

Reply:
Beleza. Dei uma lida na documentação do mongoDB para entender melhor o tipo de busca que está sendo feita. Dá uma olhada no text Score. Acho que pode ser uma boa https://docs.mongodb.com/manual/reference/operator/query/text/

Reply:
Eu to ligado, mas eu não quis usar o textscore pq tava tentando fazer com o find sem $index

Reply:
Não entendi. Por que pelo que eu entendi, a sua função weight serve para dar mais peso para certos atributos. No caso, teriamos: videoID < ownerUsername < artist < title. Contudo, o textScore serve para ordenar as palavras encontradas. Por exemplo, se eu procurar. "gg" e tivermos:
"ggeasy", "almostgg", "gg", "qualquerCoisa".

O resultado seria:
"ggeasy", "almostgg", "gg".

O textScore serve justamente para ordenar os melhores achados. Ou seja, usando o textScore viria:
"gg" > "almostgg" <=> "ggeasy". 

A <b> sugestão </b> era utilizar um em conjunto do outro. Ou seja, o textScore com os weights que você criou. Se tivesse como, seria uma boa. Se não, deixa do jeito que está.

Reply:
Deixa como tá, revisa ai pra gente aceitar. Depois a gente faz busca com substring. 👍